crocket/src/main/java/com/crocket/model/surface/SurfaceHandler.java

```diff
@@ -0,0 +1,82 @@
+package com.crocket.model.surface;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.crocket.model.entity.Ball;
+import com.crocket.shared.SurfaceType;
+
+public class SurfaceHandler {
+    private static SurfaceHandler instance = null;
+    private Set<Surface> surfaceMap;
+
+    private SurfaceHandler() {
+    }
+
+    public static SurfaceHandler getInstance() {
+        if (instance == null) {
+            instance = new SurfaceHandler();
+        }
+        return instance;
+    }
+
+    public Set<Surface> getSurfaceMap() {
+        return surfaceMap;
+    }
+
+    public void setSurfaceMap(SurfaceType[][] tilemap){
+        Set<Surface> classSurfaceMap = new HashSet<Surface>();
+        for(int i = 0; i < tilemap.length; i++){
+            for(int j = 0; j < tilemap[0].length; j++){
+                switch(tilemap[i][j]){
+                    case GRASS:
```

Comment:
This might need to be hard coded but it would be nice to have a static final constant like SURFACE_HEIGHT and SURFACE_WIDTH i think :)

Reply:
I do think that it's better to let the surfacehandler just work with whatever tilemap it gets, a surfacehandler should not need to get restricted by another class's restrictions.